src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -753,6 +811,12 @@ private protected override void SaveModel(ModelSaveContext ctx)
             ctx.Writer.Write(_predictedLabelColumnName);
             ctx.Writer.Write(_learningRate);
             ctx.Writer.Write(_classCount);
+
+            Host.AssertNonEmpty(_keyValueAnnotations);
```

Comment:
>AssertNonEmpty [](start = 17, length = 14)

You cannot assert this, since it is possible that `_keyValueAnnotations` is null: You are initializing it above in line 750 only if `inputSchema[labelColumnName].HasKeyValues()`. So you need to serialize a byte indicating whether it is null or not, followed by `_classCount` strings if it is not null. And same at deserialization time (read a byte, and then read `_classCount` strings only if that byte is `true`).

Reply:
Adjusted code with latest push so that _keyValueAnnotations will not be null, thanks to Zeeshan for the suggestion